File: client/src/app/modules/quotation/services/provision.family.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ProvisionFamilyType } from '../../quotation/model/ProvisionFamilyType';

@Injectable({
  providedIn: 'root'
})
export class ProvisionFamilyTypeService extends AppRestService<ProvisionFamilyType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getProvisionFamilyTypes() {
    return this.getListCached(new HttpParams(), "provision-family-types");
  }


  addOrUpdateProvisionFamilyType(provisionFamilyType: ProvisionFamilyType) {
    this.clearListCache(new HttpParams(), "provision-family-types");
    return this.addOrUpdate(new HttpParams(), "provision-family-type", provisionFamilyType, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/ProvisionFamilyTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.ProvisionFamilyType;

public interface ProvisionFamilyTypeService {
    public List<ProvisionFamilyType> getProvisionFamilyTypes();

    public ProvisionFamilyType getProvisionFamilyType(Integer id);

    public ProvisionFamilyType addOrUpdateProvisionFamilyType(ProvisionFamilyType provisionFamilyType);
}
